start.py

```diff
@@ -8,16 +10,42 @@
 from osfoffline.database import drop_db
 from osfoffline.gui.qt import OSFOfflineQT
 from osfoffline.utils.singleton import SingleInstance
+from osfoffline.settings import *
 
+from updater4pyi import upd_source, upd_core
+from updater4pyi.upd_iface_pyqt4 import UpdatePyQt4Interface
+from updater4pyi.upd_defs import Updater4PyiError
 
-def running_warning():
+
+def running_warning(message):
     warn_app = QApplication(sys.argv)
-    QMessageBox.information(None, 'Systray', 'OSF-Offline is already running. Check out the system tray.')
+    QMessageBox.information(None, 'Systray', message)
     warn_app.quit()
 
 
 def start():
-    SingleInstance(callback=running_warning)  # will end application if an instance is already running
+    # will end application if an instance is already running
+    SingleInstance(callback=running_warning('OSF-Offline is already running. Check out the system tray.'))
+
+    # Check for updates first and give user a way to get new version
+    try:
+        swu_source = upd_source.UpdateGithubReleasesSource(REPO)
+        swu_updater = upd_core.Updater(current_version=VERSION,
+                                       update_source=swu_source)
+        swu_interface = UpdatePyQt4Interface(updater=swu_updater,
+                                             progname=NAME,
+                                             ask_before_checking=True,
+                                             parent=QApplication.instance())
+    except Updater4PyiError:
+        pass
+
+    # Then, if the current version is too old, close the program
+    r = requests.get(MIN_VERSION_URL)
```

Comment:
This will lead to an uncaught exception if the user does not have an internet connection when the program opens. 

If the check for updates (or minimum version) fails, the user should be notified, eg by means of a dialog box.